### PR TITLE
feat: add subtract_

### LINKED ISSUE #23202: subtract_
Subtask of https://github.com/unifyai/ivy/issues/15039

### PR DESCRIPTION
**Pull request by Guilhermeslucas**
# PR Description

PR that implements subtract_ on paddle frontend.

## Related Issue

Close #23202

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Guilhermeslucas** [comment]
@D0m-inic Can I get your review in this one, since its a continuation of this https://github.com/unifyai/ivy/pull/23204 ? Thanks

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py [hunk 1]
```diff
@@ -206,3 +206,38 @@ def test_paddle_sqrt_(
         on_device=on_device,
         x=x[0],
     )
+
+
+# subtract_
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.subtract_",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py:215]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```
This should always be `valid` - if only `float` data-type is supported then that's what `valid` will return.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @Guilhermeslucas, thanks for working on this PR. Just left a minor comment and once you resolve that, make sure that all the tests are passing first and then we can merge the PR :) Happy to help out if you run into any problems. Thanks :)

**Comment by Guilhermeslucas** [comment]
Thanks for the suggestion and for the fast response @hello-fri-end . I have just accepted your suggestion

**Comment by Guilhermeslucas** [comment]
@hello-fri-end I am running into these errors. Can you please help me debug it?

image

**Comment by Guilhermeslucas** [comment]
I have fixed the issues locally. @hello-fri-end

**Comment by Guilhermeslucas** [comment]
Tests reports say that there is no failures introduced, so I imagine failures are not related to my change.

**Review by hello-fri-end — APPROVED** [review]
LGTM! Thanks for the contribution @Guilhermeslucas, merging this now :rocket:

**Comment by Guilhermeslucas** [comment]
Thank you so much for the fast review and merge @hello-fri-end

**Pull request merged**